roxie/roxiemem/roxiemem.cpp

```diff
@@ -2330,19 +2367,17 @@ class CHeap : public CInterface
     friend class HeapCompactState;
 public:
     CHeap(CChunkingRowManager * _rowManager, const IContextLogger &_logctx, const IRowAllocatorCache *_allocatorCache, unsigned _flags)
-        : logctx(_logctx), rowManager(_rowManager), allocatorCache(_allocatorCache), activeHeaplet(NULL), heaplets(NULL), flags(_flags)
+        : logctx(_logctx), rowManager(_rowManager), allocatorCache(_allocatorCache), activeHeaplet(NULL), heaplets(NULL), flags(_flags), headMaybeSpace(BLOCKLIST_NULL)
```

Comment:
@richardkchapman I think this is a style question.  Do we prefer initialisers on member variables, or initialisation within the constructor?

I think I prefer on the member variables, because it means if you add another constructor they will be initialised, and if you add a member variable you don't need to modify all the constructors.  It does look a bit strange with it mixed - should we switch all members to initialisers if we add any to a class?
An error if a member is not initialised in a constructor might also be good in the medium/long term.